src/algorithm/geometry.hxx

```diff
@@ -206,34 +261,48 @@ namespace se3
           );
     }
 
-    /// 2. Update the inner/outer objects
-    typedef GeometryModel::GeomIndexList GeomIndexList;
+    /// 2. add the collision pairs between geomModel1 and geomModel2.
+    for (Index i = 0; i < nGeom1; ++i) {
+      for (Index j = 0; j < nGeom2; ++j) {
+        geomModel1.collisionPairs.push_back(CollisionPair(i, nGeom1 + j));
+      }
+    }
+  }
+
+  inline void appendGeometryModel(GeometryModel & geomModel1,
+                                  GeometryData & geomData1,
+                                  const GeometryModel & geomModel2,
+                                  const GeometryData & geomData2)
+  {
+    /// 1&2. Call previous function.
+    appendGeometryModel(geomModel1,geomModel2);
+
+    /// 3. Update the inner/outer objects
+    Index nGeom1 = geomModel1.geometryObjects.size();
+    typedef GeometryData::GeomIndexList GeomIndexList;
     typedef std::map < JointIndex, GeomIndexList > Map_t;
-    BOOST_FOREACH(const Map_t::value_type& innerObject, geomModel2.innerObjects)
+    BOOST_FOREACH(const Map_t::value_type& innerObject, geomData2.innerObjects)
     {
-      GeomIndexList& innerGeoms = geomModel1.innerObjects[innerObject.first];
+      GeomIndexList& innerGeoms = geomData1.innerObjects[innerObject.first];
       innerGeoms.reserve(innerGeoms.size() + innerObject.second.size());
       BOOST_FOREACH(const GeomIndex& gid, innerObject.second)
       {
         innerGeoms.push_back(nGeom1 + gid);
       }
     }
-    BOOST_FOREACH(const Map_t::value_type& outerObject, geomModel2.outerObjects)
+    BOOST_FOREACH(const Map_t::value_type& outerObject, geomData2.outerObjects)
     {
-      GeomIndexList& outerGeoms = geomModel1.outerObjects[outerObject.first];
+      GeomIndexList& outerGeoms = geomData1.outerObjects[outerObject.first];
       outerGeoms.reserve(outerGeoms.size() + outerObject.second.size());
       BOOST_FOREACH(const GeomIndex& gid, outerObject.second)
       {
         outerGeoms.push_back(nGeom1 + gid);
       }
     }
 
-    /// 3. add the collision pairs between geomModel1 and geomModel2.
-    for (Index i = 0; i < nGeom1; ++i) {
-      for (Index j = 0; j < nGeom2; ++j) {
-        geomModel1.collisionPairs.push_back(CollisionPair(i, nGeom1 + j));
-      }
-    }
+    // FIXME: I copy here the previous algo. Maybe it would be safer to simply use the
+    // fillInnerOuterObjectMaps() method (however marginally less efficient: is it important?).
```

Comment:
I would say it's rather dangerous to keep a function that doesn't do the
job it's is meant to do. Creating a new data from geomModel1 solves the
issue, right? If yes, removing is better.

Joseph Mirabel

Le 8 sept. 2016 09:50, "Nicolas Mansard" <notifications@github.com> a
écrit :

> In src/algorithm/geometry.hxx
> <https://github.com/stack-of-tasks/pinocchio/pull/314#discussion_r77960073>
> :
>
> >        outerGeoms.reserve(outerGeoms.size() + outerObject.second.size());
> >        BOOST_FOREACH(const GeomIndex& gid, outerObject.second)
> >        {
> >          outerGeoms.push_back(nGeom1 + gid);
> >        }
> >      }
> >
> > -    /// 3. add the collision pairs between geomModel1 and geomModel2.
> > -    for (Index i = 0; i < nGeom1; ++i) {
> > -      for (Index j = 0; j < nGeom2; ++j) {
> > -        geomModel1.collisionPairs.push_back(CollisionPair(i, nGeom1 + j));
> > -      }
> > -    }
> > +    // FIXME: I copy here the previous algo. Maybe it would be safer to simply use the
> > +    // fillInnerOuterObjectMaps() method (however marginally less efficient: is it important?).
>
> Should I simply remove this second function?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/stack-of-tasks/pinocchio/pull/314/files/ae610919c80f9cddcc1023df7fe29686a975b0c8#r77960073>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACRw8R-K3qzfp99VqsBNFRywqFkXjjwSks5qn75MgaJpZM4J23-P>
> .
>